node-data/src/message.rs

```diff
@@ -493,6 +493,20 @@ pub mod payload {
         NoQuorum = 3,
     }
 
+    impl Vote {
+        pub fn size(&self) -> usize {
+            const ENUM_BYTE: usize = 1;
+
+            let data_size: usize = match &self {
+                Vote::NoCandidate => 0,
+                Vote::Valid(_) => 32,
+                Vote::Invalid(_) => 32,
+                Vote::NoQuorum => 0,
+            };
```

Comment:
AFAIK, Rust allocates the size based on the maximum value the enum can take. So even if it's NoCandidate, the enum's size is still 32+1.

Reply:
Also, to avoid mistakes, can you use `size_of`?

Reply:
We are not talking about the rust memory allocation, we are talking about the size of the message sent (so in its serialized form)

Reply:
Agreed on using the serialized size